### ISSUE TITLE
Command parsing fails in v4.31.0

### ISSUE THREAD

**Issue by roel-koemeester** [inspec/inspec#5459]
When running first example test from https://docs.chef.io/inspec/resources/interfaces/ it fails in version 4.31.0

# Describe the problem

Given this test: 
describe interfaces do
  its('names') { should include 'eth0' }
end

and given a system that has interfaces "eth0" and "lo"
on version 4.28.0 it succeeds and on version 4.31.0 it results in:
expected ["eth0  lo\r"] to include "eth0"

**Comment by clintoncwolfe** [comment]
Thanks for raising the issue! What platform was this on?

**Comment by roel-koemeester** [comment]
This was ran on centos7

**Comment by clintoncwolfe** [comment]
Confirmed on CentOS 8
```
inspec> interfaces.names
=> ["eth0  lo\r"]
```

Appears to only occur over SSH transport. Line endings change on SSH commands when a timeout is used.

Bisected to introduced on 4.30.0, #5443.

**Comment by deric4** [comment]
This seems to be affecting quite a few controls in https://github.com/dev-sec/cis-dil-benchmark across multiple distros, requiring us to pin the last known working version to `4.29.3` for now.

One example is  [cis-dil-benchmark-1.1.21](https://github.com/dev-sec/cis-dil-benchmark/blob/77785439eac5d2a56e2e002c5bcdf2d214985ff9/controls/1_1_filesystem_configuration.rb#L423-L434) where the `\` character gets stripped from command string causing inspec to execute an invalid command.

```ruby
control 'cis-dil-benchmark-1.1.21' do
  title 'Ensure sticky bit is set on all world-writable directories'
  desc  "Setting the sticky bit on world writable directories prevents users from deleting or renaming files in that directory that are not owned by them.\n\nRationale: This feature prevents the ability to delete or rename files in world writable directories (such as /tmp) that are owned by another user."
  impact 1.0

  tag cis: 'distribution-independent-linux:1.1.21'
  tag level: 1

  describe command("df --local -P | awk '{ if (NR!=1) print $6 }' | xargs -I '{}' find '{}' -xdev -type d \( -perm -0002 -a ! -perm -1000 \)") do
    its('stdout') { should cmp '' }
  end
end
```

**Comment by clintoncwolfe** [comment]
The issue here is that the SSH transport code was allocating a PTY when a timeout was specified (which since #5443, one always is, since it now defaults to 3600 seconds). That alters the semantics of several shell commands, including `ls` and apparently the `\` character. There must be numerous command strings that would be affected. 
https://github.com/inspec/train/pull/676 fixes the issue, I tested the command invocation `"df --local -P | awk '{ if (NR!=1) print $6 }' | xargs -I '{}' find '{}' -xdev -type d \( -perm -0002 -a ! -perm -1000 \)"`

**Comment by clintoncwolfe** [comment]
I'm going to ahead and close this, since we've released a fix for this aspect of the issue in inspec 4.32.0. We do have another related outstanding issue, inspec/train#677
